src/clientdlg.cpp

```diff
@@ -608,11 +603,8 @@ void CClientDlg::closeEvent ( QCloseEvent* Event )
     ConnectDlg.close();
     AnalyzerConsole.close();
 
-    // if connected, terminate connection
-    if ( pClient->IsRunning() )
-    {
-        pClient->Stop();
-    }
+    // Disconnect if needed
```

Comment:
I'd do this as the last thing before `Event->accept();`

And do we want to ignore the triggered `CClient::Disconnected` signal that will happen when `CClient::Stop` gets called?

Reply:
> And do we want to ignore the triggered CClient::Disconnected signal that will happen when CClient::Stop gets called?

Probably not.

Reply:
> I'd do this as the last thing before Event->accept();

This gives a small timeframe (in theory) where the fader settings are not stored, but the client disconnected. I don't think that's a problem.